utils/ccdatofhir/mapping_datatypes.bal

```diff
@@ -0,0 +1,215 @@
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// --------------------------------------------------------------------------------------------#
+// Source C-CDA to FHIR - Datatype Mappings
+// --------------------------------------------------------------------------------------------#
+
+import ballerina/log;
+import ballerinax/health.fhir.r4;
+
+xmlns "urn:hl7-org:v3" as v3;
+
+# Map C-CDA id to FHIR Identifier.
+#
+# + idElement - C-CDA id element
+# + return - Return FHIR Identifier
+public isolated function mapCcdaIdToFhirIdentifier(xml idElement) returns r4:Identifier? {
+    string|error? idVal = idElement.extension;
+    string|error? rootVal = idElement.root;
+    if rootVal !is string {
+        log:printDebug("Mandatory field root not available", rootVal);
+        return ();
+    }
+    if rootVal.equalsIgnoreCaseAscii("2.16.840.1.113883.4.1") && idVal is string {
+        return {
+            system: "http://hl7.org/fhir/sid/us-ssn",
+            value: idVal
+        };
+    } else if idVal is string {
+        return {
+            system: rootVal,
+            value: string `urn:oid:${idVal}`
+        };
+    }
+    return {
+        system: "urn:ietf:rfc:3986",
+        value: rootVal
+    };
+}
+
+# Map C-CDA address to FHIR Address.
+#
+# + addressElement - C-CDA address element
+# + return - Return FHIR Address
+public isolated function mapCcdaAddressToFhirAddress(xml addressElement) returns r4:Address? {
+    xml streetAddressLineElement = addressElement/<v3:streetAddressLine|streetAddressLine>;
+    xml cityElement = addressElement/<v3:city|city>;
+    xml stateElement = addressElement/<v3:state|state>;
+    xml postalCodeElement = addressElement/<v3:postalCode|postalCode>;
+
+    string streetAddressLine = streetAddressLineElement.data().trim();
+    string city = cityElement.data().trim();
+    string state = stateElement.data().trim();
+    string postalCode = postalCodeElement.data().trim();
+
+    if streetAddressLine != "" || city != "" || state != "" || postalCode != "" {
+        return {
+            line: [streetAddressLine],
+            city: city,
+            state: state,
+            postalCode: postalCode
+        };
+    }
+    log:printDebug("Address fields not available");
+    return ();
+}
+
+# Map C-CDA telecom to FHIR ContactPoint.
+#
+# + telecomElement - C-CDA telecom element
+# + return - Return FHIR ContactPoint
+public isolated function mapCcdaTelecomToFhirTelecom(xml telecomElement) returns r4:ContactPoint? {
+    string|error? telecomUse = telecomElement.use;
+    string|error? telecomValue = telecomElement.value;
+
+    string? valueVal = ();
```

Comment:
with the change in https://github.com/ballerina-platform/module-ballerinax-health.fhir.r4/pull/16/files#r1261096563 you will not need `?`